docs/testing.md

```diff
@@ -0,0 +1,187 @@
+* [Test harness usage](#test-harness-usage)
+   * [Framework test suite](#framework-test-suite)
+      * [Start a Kubernetes cluster with kind (optional)](#start-a-kubernetes-cluster-with-kind-optional)
+      * [Run the Framework tests](#run-the-framework-tests)
+   * [Integration test suite](#integration-test-suite)
+* [Writing test cases](#writing-test-cases)
+   * [Test case directory structure](#test-case-directory-structure)
+   * [Test steps](#test-steps)
+      * [Test assertions](#test-assertions)
+         * [Advanced test assertions](#advanced-test-assertions)
+* [Further Reading](#further-reading)
+
+## Test harness usage
+
+There are two official KUDO test suites:
+
+* The [Framework test suite](#framework-test-suite) which tests Frameworks built on KUDO. These tests live in the Frameworks repository.
+* The [integration test suite](#integration-test-suite) which tests the KUDO controller itself.
+
+Both are run using the KUDO test harness as a part of our CI processes.
+
+### Framework test suite
+
+The Framework test suite is written and run by Framework developers and CI to test that Frameworks work correctly.
+
+
+#### Start a Kubernetes cluster with kind (optional)
+
+The Framework tests rely on a functioning Kubernetes cluster. As the test harness can run KUDO, it is not necessary to install KUDO prior to running the tests.
+
+If you do not have a Kubernetes cluster to use for the tests, you can start one with [kind](https://github.com/kubernetes-sigs/kind):
+
+There is a Make target in the [Frameworks repository](https://github.com/kudobuilder/frameworks) that you can use to start it:
+
+```
+make create-cluster
+export KUBECONFIG=$(bin/kind get kubeconfig-path)
+```
+
+Now that the cluster is running, you can use it as a testing environment.
+
+#### Run the Framework tests
+
+To run the Framework test suite, run the following in the [Frameworks repository](https://github.com/kudobuilder/frameworks):
+
+```
+make test
+```
+
+Framework test suites are stored in the `tests` subdirectory of each [Framework](https://github.com/kudobuilder/frameworks/tree/master/repository), e.g.:
+
+```
+./repository/zookeeper/tests/
+./repository/mysql/tests/
+```
+
+Every `Framework` and `FrameworkVersion` is installed into the cluster by `make test` prior to running the test suite.
+
+### Integration test suite
+
+The [integration test suite](https://github.com/kudobuilder/kudo/tree/master/pkg/test/test_data) lives in `pkg/test/test_data` and is intended for testing KUDO functionality (rather than any end-user Frameworks).
+
+No Kubernetes cluster is required, as the integration tests use the [controller-runtime's envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest).
+
+To run the KUDO integration test suite, run the following in your [local KUDO checkout](https://github.com/kudobuilder/kudo):
+
+```
+make test
+```
+
+The resources under `config/samples/test-framework/` are installed prior to running the tests.
```

Comment:
This is switching back to writing framework tests, where as in the last command was running kudo test.
I think, a differentiation is required.